### PR TITLE
*: update version and check version before starting

### PR DESCRIPTION
**Pull request by overvenus**

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [root]
 name = "tikv"
-version = "0.0.1"
+version = "0.9.0-pre-ga"
```
Comment (andelf) [inline Cargo.lock]: why 0.9.0? if GA is 1.0.0?

Comment (siddontang) [inline Cargo.lock]: yes, we decide pre-ga is 0.9.0, ga is 1.0.0.

Comment (siddontang) [inline Cargo.lock]: but I prefer using 0.9 in #2054, not here.

Comment (andelf) [inline Cargo.lock]: 🤣 funny versioning scheme, makes software release life cycle a joke.
Better give it up this time and set it up at/after 1.0.0, i.e. GA

File: src/util/mod.rs [hunk 2]
```diff
@@ -454,15 +461,86 @@ pub fn is_even(n: usize) -> bool {
...
+
+// It fails when the the major versions are different.
+fn check_version_compatiblity(previous_ver: &str, current_ver: &str) -> Result<(), String> {
+    let previous_ver = Version::parse(previous_ver)
+        .map_err(|e| format!("{:?}", e))?;
+    let current_ver = Version::parse(current_ver).map_err(|e| format!("{:?}", e))?;
+    if previous_ver.major == current_ver.major {
+        Ok(())
+    } else {
+        Err(format!(
```
Comment (BusyJay) [inline src/util/mod.rs]: I think we should allow different major version here. For example, we may increase major version when we introduce some great features. But it doesn't mean that the changes are incompatible.

/cc @siddontang @ngaut

Comment (siddontang) [inline src/util/mod.rs]: If we follow Semantic Versioning standard, the major version is for incompatibility change after we release 1.0.0.

File: Cargo.toml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "tikv"
-version = "0.0.1"
+version = "0.9.0-pre-ga"
```
Comment (siddontang) [inline Cargo.toml]: Below 1.0.0, we may also meet incompatibility, E.g, latter 0.9.0 for real dedicated raft RocksDB.

So I prefer using 0.8.0 here, not 0.9.0.

File: src/util/mod.rs [hunk 4]
```diff
@@ -454,15 +461,86 @@ pub fn is_even(n: usize) -> bool {
...
+    }
+}
+
+// It fails when the the major versions are different.
+fn check_version_compatiblity(previous_ver: &str, current_ver: &str) -> Result<(), String> {
+    let previous_ver = Version::parse(previous_ver)
+        .map_err(|e| format!("{:?}", e))?;
+    let current_ver = Version::parse(current_ver).map_err(|e| format!("{:?}", e))?;
+    if previous_ver.major == current_ver.major {
+        Ok(())
```
Comment (javaforfun) [inline src/util/mod.rs]: do we need update version_lock after we run new version on old data?

**Pull request closed without merging**